### PR TITLE
Promotion from AMD internal branch for 2023.Q1.3

### PR DESCRIPTION
**Pull request by WenqingLiAMD**

### CODE REVIEW

File: lgc/CMakeLists.txt [hunk 1]
```diff
@@ -81,6 +81,7 @@ if(NOT EXISTS ${CMAKE_CURRENT_BINARY_DIR}/state)
     file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/state")
 endif()
 tablegen(LGC state/LgcDialect.cpp.inc -gen-dialect-defs --dialect lgc
+    -no-memory-effects # TODO: Remove this once all relevant LLVM branches have the new memory(...) attribute
```
Comment (nhaehnle) [inline lgc/CMakeLists.txt:84]: Please remove this change. It was added as a fix for the fact that some of the internal builds used a relatively old version of LLVM still. Actually, we should probably see about remove it in the internal version as well.

**Review by nhaehnle — COMMENTED** [review]
One issue noted in a comment, the rest looks reasonable to me.

File: lgc/CMakeLists.txt (hunk 1 above)
Comment (WenqingLiAMD) [reply lgc/CMakeLists.txt]: Updated, thanks~

File: llpc/test/shaderdb/relocatable_shaders/PipelineVsFs_GlueCache.pipe [hunk 2]
```diff
@@ -118,6 +118,6 @@ binding[0].binding = 1
 binding[0].stride = 16
 binding[0].inputRate = VK_VERTEX_INPUT_RATE_VERTEX
 attribute[0].location = 0
-attribute[0].binding = 0
+attribute[0].binding = 1
```
Comment (Flakebi) [inline llpc/test/shaderdb/relocatable_shaders/PipelineVsFs_GlueCache.pipe:121]: This change seems to cause a test failure

**Comment by amdvlk-admin** [comment]
### Test summary for commit 65b11e15d35c175b40e43aee5621dc0fcd98a57f
<details><summary>CTS tests (Failed: 0/207173)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 36991/69071 (53.6%)</li>
<li>Failed: 0/69071 (0.0%)</li>
<li>Not Supported: 32080/69071 (46.4%)</li><li>Warnings: 0/69071 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 37209/69039 (53.9%)</li>
<li>Failed: 0/69039 (0.0%)</li>
<li>Not Supported: 31830/69039 (46.1%)</li><li>Warnings: 0/69039 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 34229/69063 (49.6%)</li>
<li>Failed: 0/69063 (0.0%)</li>
<li>Not Supported: 34834/69063 (50.4%)</li><li>Warnings: 0/69063 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/test/shaderdb/relocatable_shaders/PipelineVsFs_GlueCache.pipe (hunk 2 above)
Comment (WenqingLiAMD) [reply llpc/test/shaderdb/relocatable_shaders/PipelineVsFs_GlueCache.pipe:121]: All tests pass now, thanks

**Comment by amdvlk-admin** [comment]
### Test summary for commit 49b5280d33037aaa8802dbc98774571fc2514513
<details><summary>CTS tests (Failed: 0/207358)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 36990/69126 (53.5%)</li>
<li>Failed: 0/69126 (0.0%)</li>
<li>Not Supported: 32136/69126 (46.5%)</li><li>Warnings: 0/69126 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 37210/69104 (53.8%)</li>
<li>Failed: 0/69104 (0.0%)</li>
<li>Not Supported: 31894/69104 (46.2%)</li><li>Warnings: 0/69104 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 34228/69128 (49.5%)</li>
<li>Failed: 0/69128 (0.0%)</li>
<li>Not Supported: 34900/69128 (50.5%)</li><li>Warnings: 0/69128 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request closed without merging**